packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/CameraUtils.java

```diff
@@ -186,6 +160,23 @@ public static Messages.PlatformFocusMode focusModeToPigeon(@NonNull FocusMode fo
     return Messages.PlatformFocusMode.AUTO;
   }
 
+  /**
+   * Converts a PlatformFocusMode from Pigeon to a FocusMode from the autofocus package.
+   *
+   * @param focusMode A PlatformFocusMode.
+   * @return The corresponding FocusMode.
+   */
+  @Nullable
+  public static FocusMode focusModeFromPigeon(@NonNull Messages.PlatformFocusMode focusMode) {
+    switch (focusMode) {
+      case AUTO:
+        return FocusMode.auto;
+      case LOCKED:
+        return FocusMode.locked;
+    }
+    return null;
```

Comment:
This can throw since it can't happen.

Reply:
I currently modified these methods to throw `IllegalStateException` to represent the unreachable code.